Jenkinsfile

```diff
@@ -60,12 +65,37 @@ pipeline {
                 not { branch 'develop' }
                 not { branch 'master' }
             }
-            steps { 
+            steps {
                 script {
                     utils.killOldBuilds()
                 }
             }
         }
+        stage("Make-doxygen") {
+            agent any
+            when { branch 'PR-795'}
+            steps {
+                    retry(3) { checkout scm }
+                    withCredentials([usernamePassword(credentialsId: 'a630aebc-6861-4e69-b497-fd7f496ec46b',
+                        usernameVariable: 'GIT_USERNAME', passwordVariable: 'GIT_PASSWORD')]) {
+                        sh """#!/bin/bash
+                            set -x
+                            make doxygen
+                            git config --global user.email "mc.stanislaw@gmail.com"
+                            git config --global user.name "Stan Jenkins"
+                            git checkout --detach
+                            git branch -D gh-pages
+                            git push https://${GIT_USERNAME}:${GIT_PASSWORD}@github.com/${fork()}/math.git :gh-pages
```

Comment:
If this is only ever running on master, maybe we don't need the `${fork()}` here and below, we can just use stan-dev? Don't feel strongly either way.

Reply:
Yes I'll change this to stan-dev